### PR TITLE
Update Stylelint in peer dependency

### PR DESCRIPTION
**Pull request by ai**
Migration to Stylelint 16 should not be hard

https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-16.md

### CODE REVIEW

**Comment by ai** [comment]
Hm. Right now we have a warning:

```
CommonJS plugins are deprecated ("/home/ai/Dev/slowreader/node_modules/.pnpm/stylelint-use-logical@2.1.0_stylelint@16.0.1/node_modules/stylelint-use-logical/index.cjs"). See https://stylelint.io/migration-guide/to-16
```

And error:

```
Unknown rule csstools/use-logical  csstools/use-logical
```

**Comment by ai** [comment]
Seems like it happens because this repo is using Rollup to compile ESM to CJS because [`stylelint-gamut`](https://github.com/fpetrakov/stylelint-gamut/) (which is also written on pure CJS) has no problems.

**Comment by ybiquitous** [comment]
Hi, I'm a maintainer of Stylelint. The `Unknown rule csstools/use-logical` error is caused by the incorrect exporting `ruleName`:

https://github.com/csstools/stylelint-use-logical/blob/faea20dfc1a74c620b7e39522a185aa8ee96c354/index.js#L145

Although this code works with Stylelint v15, it's an undocumented behavior. See the [v15 plugin documentation](https://github.com/stylelint/stylelint/blob/15.11.0/docs/developer-guide/plugins.md).

I strongly recommend fixing this plugin as follows. It works with both v15 and v16:

```diff
--- a/index.js
+++ b/index.js
@@ -7,7 +7,7 @@ import walk from './lib/walk';
 
 const reportedDecls = new WeakMap();
 
-export default stylelint.createPlugin(ruleName, (method, opts, context) => {
+function ruleFunc(method, opts, context) {
 	const propExceptions = [].concat(Object(opts).except || []);
 	const isAutofix = isContextAutofixing(context);
 	const dir = /^rtl$/i.test(Object(opts).direction) ? 'rtl' : 'ltr';
@@ -140,9 +140,11 @@ export default stylelint.createPlugin(ruleName, (method, opts, context) => {
 			});
 		}
 	};
-});
+}
 
-export { ruleName }
+ruleFunc.ruleName = ruleName;
+
+export default stylelint.createPlugin(ruleName, ruleFunc);
 
 const isMethodIndifferent = method => method === 'ignore' || method === false || method === null;
 const isMethodAlways = method => method === 'always' || method === true;
```

**Comment by JounQin** [comment]
Is that a good practice to add properties onto function?

**Comment by ybiquitous** [comment]
> Is that a good practice to add properties onto function?

I'm unsure if that's a good practice or not. But it respects compatibility.

**Comment by ai** [comment]
@JounQin sure. JS allows it and I do not know big disadvantages.

**Comment by JounQin** [comment]
I think we should add back the previous support for compatibility, and export `default` + `ruleName` should be preferred IMO.

**Comment by JounQin** [comment]
> @JounQin sure. JS allows it and I do not know big disadvantages.

It doesn't work well with TypeScript by default.

**Comment by ai** [comment]
> I think we should add back the previous support for compatibility, and export default + ruleName should be preferred IMO.

We should move the discussion to Stylelint repo. Anyway this is a Stylelint current API.

**Comment by ai** [comment]
@jonathantneal I updated PR with @ybiquitous  fixes

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "stylelint-tape": "3.0.0"
   },
   "peerDependencies": {
-    "stylelint": ">= 11 < 16"
+    "stylelint": ">= 11 < 17"
```
Comment (JounQin) [inline package.json:45]: The new plugin structure works on `>=11 <15`? @ybiquitous

Comment (ybiquitous) [reply package.json:45]: It's not a **new** structure.

Comment (JounQin) [reply package.json:45]: OK, so it's still compatible here, right?

Comment (ybiquitous) [reply package.json:45]: Yes. `export { ruleName }` happens to work.

Comment (JounQin) [reply package.json:45]: https://github.com/csstools/stylelint-use-logical/pull/22/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R146

@ybiquitous It is not exported anymore.

Comment (JounQin) [reply package.json:45]: @ybiquitous Any thoughts?

Comment (ai) [reply package.json:45]: The Stylelint API was not changed, we should not change this range.

Comment (JounQin) [reply package.json:45]: > The Stylelint API was not changed, we should not change this range.

I'm just confirming whether current pattern also works on `>=11 <15`.

> We just moved to a correct API which worked before by accident and stop working in Stylelint 16.

That's why I said for compatibility, stylelint should not remove the API, and the API is better than current one IMO.

**Comment by inomdzhon** [comment]
Hi, there!

Any updates?

**Comment by romainmenke** [comment]
No updates yet, but I haven't forgotten.
Requires time to sort out publishing access :)

**Comment by inomdzhon** [comment]
> No updates yet, but I haven't forgotten. Requires time to sort out publishing access :)

Hi, again)

Do you have any news about access?

**Comment by romainmenke** [comment]
Thank you @ai, and everyone else 🙇

**Comment by romainmenke** [comment]
This has been released as `2.1.1`

**Comment by ramil-k** [comment]
Thanks a lot for your work.

I've just migrated to 2.1.1 and I still getting an error message: 
```CommonJS plugins are deprecated ("/home/ai/Dev/slowreader/node_modules/.pnpm/stylelint-use-logical@2.1.0_stylelint@16.0.1/node_modules/stylelint-use-logical/index.cjs"). See https://stylelint.io/migration-guide/to-16```

Is there some configurations that I should update?

**Comment by romainmenke** [comment]
That message is kinda expected.
Stylelint is in the process of migrating to pure esm
In a future update we will update this plugin to pure esm.

But at this time it is only a warning and shouldn't be to much of a nuisance :)

**Comment by ramil-k** [comment]
Oh, I see, thanks for clarification. It works.

**Pull request merged**